### DISCUSSION TITLE
Motifs

### DISCUSSION THREAD

**Discussion by Haaroon** [Pometry/Raphtory#841] [Bounty Board]
# Motifs
* 1 node 2-edge motifs :heavy_check_mark: 
* Clustering coefficient :heavy_check_mark: 
* I/O motifs (e.g. for transaction nets) :heavy_check_mark: (for motifs, one usually compares their number with that of aa properly randomised system, using e.g. a z score) 
* Reciprocity of edges :heavy_check_mark: 
* 3-node motifs :heavy_check_mark:

**Comment by allan0095** [comment]
Is there an easy way of finding the mapping to motifs?

This is the only way I have found it https://github.com/Pometry/Raphtory/blob/85a9eab8839cc2e602f2737897e48de8653ad967/python/python/raphtory/plottingutils.py#L97

**Reply by miratepuffin** [reply]
Hi @allan0095, I believe that is the only place we have it codified, but I can check with the algorithm author for you.

**Reply by allan0095** [reply]
I was mainly confused how a 40 length array is mapping to a 6x6 matrix. Seems like the two node notifs are double counted?

**Reply by narnolddd** [reply]
Hi @allan0095! Good point, when the global motifs are counted, the 8 lots of two-node motifs can really be considered as 4 as each of the two node motif counts is mirrored in a sense (for example an "all-outgoing" two-node motif for one node is an "all-incoming" motif for the other node). You can see this when you run global motifs, the two-node ones should be symmetric about the 4th element. In this way, the 6x6 motif matrix is only really good for plotting global motif counts. The mapping I used was the same as the ordering here: 
![image](https://github.com/user-attachments/assets/7c485512-8ddb-4c22-bfa4-4cf426ec8fe2)
but I will try and make some time to add this into the docs as I agree this should be made clearer. Hope this helps a bit?

**Reply by allan0095** [reply]
Thanks, I was able to come to the same conclusion after looking at the mapping line by line. In terms of making it clearer, adding the mapping list into the docs or removing the duplicates would help.
